pkg/policy/api/ip_protocol.go

```diff
@@ -0,0 +1,38 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package api
+
+// IPProtocols is a list of IP protocols that are allowed. This field is intended for protocols
+// that operate directly over IP without transport-layer ports (unlike TCP or UDP).
+type IPProtocols []IPProtocol
```

Comment:
Maybe this could implement `pkg/policy/api.PortsIterator`? Then we could treat it the same as port+proto and icmp?